protop-core/src/main/java/io/protop/core/sync/SyncService.java

```diff
@@ -121,16 +143,33 @@ public SyncService(AuthService<?> authService,
     /**
      * Filter the children directories and files and add to the given tree.
      */
-    private void filterValidChildren(DirectoryWithFiles directoryWithFiles, File path) throws FileAlreadyExistsException {
+    private void filterValidChildren(
+            final DirectoryWithFiles directoryWithFiles,
+            final File project) throws FileAlreadyExistsException {
+        final Path rootDir = Manifest
+                .from(project)
+                .map(Manifest::getRootDir)
+                .orElse(Path.of(""));
+
+        filterValidChildren(directoryWithFiles, project, rootDir);
+    }
+
+    @VisibleForTesting
+    void filterValidChildren(
+            final DirectoryWithFiles directoryWithFiles,
+            final File path,
+            final Path rootDir) throws FileAlreadyExistsException {
         if (path.isDirectory()) {
-            File[] children = Optional.ofNullable(path.listFiles())
+            final File[] children = Optional
+                    .ofNullable(path.listFiles())
```

Comment:
```suggestion
            final File[] children = Optional.ofNullable(path.listFiles())
```

I generally prefer newlines _after_ chained methods, not before. e.g.

```java
fn.a()
  .b()
```
and not
```java
fn
  .a()
  .b()
```
```